contracts/SiphonDepositor.sol

```diff
@@ -136,9 +144,18 @@ contract SiphonDepositor {
      * @dev Convert rAURA for AURA at the pro rata rate
      */
     function convert(uint256 _amount, bool _lock) external {
-        // TODO: implement lock functionality and penalty
         uint256 amountOut = getAmountOut(_amount);
-        rCvx.transferFrom(msg.sender, address(this), _amount);
-        cvx.transfer(msg.sender, amountOut);
+
+        if (_lock) {
+            cvx.safeApprove(address(auraLocker), 0);
+            cvx.safeApprove(address(auraLocker), amountOut);
+            auraLocker.lock(msg.sender, amountOut);
+        } else {
+            // If there is an address for auraLocker, and not locking, apply a penalty
+            uint256 penalty = amountOut * penaltyBp / 1000;
+            uint256 amountWithPenalty = amountOut - penalty;
+            rCvx.transferFrom(msg.sender, address(this), _amount);
```

Comment:
Could just burn this. The SiphonDepositor could be `ERC20` itself, and then wouldn't even need to give approval. Also, need to burn if `lock` == true also.